test/kubemark/iks/startup.sh

```diff
@@ -0,0 +1,297 @@
+  sed -i'' -e "s/{{use_real_proxier}}/${USE_REAL_PROXIER}/g" "${RESOURCE_DIRECTORY}/hollow-node.yaml"
+  sed -i'' -e "s'{{kubemark_mig_config}}'${KUBEMARK_MIG_CONFIG:-}'g" "${RESOURCE_DIRECTORY}/hollow-node.yaml"
+  "${KUBECTL}" create -f "${RESOURCE_DIRECTORY}/hollow-node.yaml" --namespace="kubemark"
+
+  echo "Created secrets, configMaps, replication-controllers required for hollow-nodes."
+}
+
+# Wait until all hollow-nodes are running or there is a timeout.
+function wait-for-hollow-nodes-to-run-or-timeout {
+  echo -n "Waiting for all hollow-nodes to become Running"
+  start=$(date +%s)
+  nodes=$("${KUBECTL}" --kubeconfig="${KUBECONFIG}" get node 2> /dev/null) || true
+  ready=$(($(echo "${nodes}" | grep -v "NotReady" | wc -l) - 1))
+  
+  until [[ "${ready}" -ge "${NUM_REPLICAS}" ]]; do
+    echo -n "."
+    sleep 1
+    now=$(date +%s)
+    # Fail it if it already took more than 30 minutes.
+    if [ $((now - start)) -gt 1800 ]; then
+      echo ""
+      echo -e "${color_red} Timeout waiting for all hollow-nodes to become Running. ${color_norm}"
+      # Try listing nodes again - if it fails it means that API server is not responding
+      if "${KUBECTL}" --kubeconfig="${KUBECONFIG}" get node &> /dev/null; then
+        echo "Found only ${ready} ready hollow-nodes while waiting for ${NUM_NODES}."
+      else
+        echo "Got error while trying to list hollow-nodes. Probably API server is down."
+      fi
+      spawn-config
+      pods=$("${KUBECTL}" get pods -l name=hollow-node --namespace=kubemark) || true
+      running=$(($(echo "${pods}" | grep "Running" | wc -l)))
+      echo "${running} hollow-nodes are reported as 'Running'"
+      not_running=$(($(echo "${pods}" | grep -v "Running" | wc -l) - 1))
+      echo "${not_running} hollow-nodes are reported as NOT 'Running'"
+      echo $(echo "${pods}" | grep -v "Running")
+      exit 1
+    fi
+    nodes=$("${KUBECTL}" --kubeconfig="${KUBECONFIG}" get node 2> /dev/null) || true
+    ready=$(($(echo "${nodes}" | grep -v "NotReady" | wc -l) - 1))
+  done
+  echo -e "${color_green} Done!${color_norm}"
+}
+
+############################### Main Function ########################################
+# In order for the cluster autoscalar to function, the template file must be changed so that the ":443"
+# is removed. This is because the port is already given with the MASTER_IP.
+
+
+# Create clusters and populate with hollow nodes
+complete-login
+build-kubemark-image
+choose-clusters
+generate-values
+set-hollow-master
+echo "Creating kube hollow node resources"
+create-kube-hollow-node-resources
+master-config
+echo -e "${color_blue}EXECUTION COMPLETE${color_norm}"
+
+# Check status of Kubemark
+echo -e "${color_yellow}CHECKING STATUS${color_norm}"
+wait-for-hollow-nodes-to-run-or-timeout
+
+# Celebrate
+echo ""
+echo -e "${color_blue}SUCCESS${color_norm}"
+clean-repo
+exit 0
```

Comment:
Please remove this unneeded exit command and the extra trailing newlines.

Reply:
Isn't the exit command necessary, so the rest of start-kubemark.sh is not run? Thats the purpose of the custom startup.

Reply:
Yeah, that makes sense. I was originally under the impression you were gonna do it from where you call this script.